### PR TITLE
Adding "smooth_l1_loss" to PyTorch frontend

### PR DESCRIPTION
**Pull request by YueBro**
Adding `smooth_l1_loss` to PyTorch frontend.

### CODE REVIEW

**Comment by YueBro** [comment]
Close #5546

**Comment by aditya-pola** [comment]
Hey @YueBro, this looks great! Just require a few changes before I can approve,
- Please add typehints for all parameters of the funtion.
- Your implementation of Smooth L1 loss is incorrect. You can find the accurate formula here: [Pytorch docs for Smooth L1 loss](https://pytorch.org/docs/stable/generated/torch.nn.SmoothL1Loss.html)

**Comment by YueBro** [comment]
Hi @aditya-pola , I have fixed the problems you've mentioned, and please take a look.

Please ignore the changes done on `data_type.py` when merging, as I'll open a new issue and pr later on.

**Comment by WOWspring** [comment]
Hello, I have a question: I am making contribution to "smooth_l1_loss" for weeks, and I'm still working on it. It seems that we are doing the same thing?

**Comment by aditya-pola** [comment]
Hello @WOWspring, thank you for bringing this issue to our attention. We've noticed that you did not link your PR ([Point (d) of ToDo List Issues for future reference](https://lets-unify.ai/ivy/contributing/1_the_basics.html#todo-list-issues)) with the issue which has led to your issue being marked [stale](https://github.com/unifyai/ivy/issues/3532) and it's subsequent closure, which is when it was made available to @YueBro. 

We value your contribution and want to merge your work as well, so I suggest the following:
@YueBro please consider this issue complete and I will merge your frontend implementation. Do not implement the test for the function.
@WOWspring please remove the implementation of the frontend from your PR as we will accept @YueBro 's version. @WOWspring please implement the test for @YueBro 's version of the frontend by pulling from master after I have merged their PR (#5563)

To sum it up, @YueBro implement the frontend function, @WOWspring implement the test for their frontend function.

I will merge this PR after I receive acknowledgement from both of you. If either of you are submitting these PRs for the internship task, please rest assured that it will not have a bearing on your application.

**Comment by YueBro** [comment]
@aditya-pola , I'm totally fine with this suggestion.

@WOWspring , sorry for that, I didn't know that this task is taken.

**Comment by WOWspring** [comment]
@aditya-pola , I'm fine with this suggestion as well, sorry for my negligence in work. So now I juest need to wait for your mergence, right?

@YueBro I also made mistakes in my work, which caused you trouble.

**Comment by aditya-pola** [comment]
No worries, I'm glad we got this sorted out! :)

**Comment by kurshakuz** [comment]
Added missing test in https://github.com/unifyai/ivy/pull/7494
